bindings/pyroot/pythonizations/python/ROOT/pythonization/_tdirectory.py

```diff
@@ -8,6 +8,48 @@
 # For the list of contributors see $ROOTSYS/README/CREDITS.                    #
 ################################################################################
 
+r'''
+/**
+\class TDirectory
+\brief \parblock \endparblock
+\htmlonly
+<div class="pyrootbox">
+\endhtmlonly
+## PyROOT
+
+From Python, it is possible to inspect the content of a TDirectory object
+as if the subdirectories and objects it contains were its attributes.
+Moreover, once a subdirectory or object is accessed for the first time,
+it is cached for later use.
+For example, assuming `d` is a TDirectory instance:
+\code{.py}
+# Access a subdirectory
+d.subdir
+
+# We can go further down in the hierarchy of directories
+d.subdir.subsubdir
+
+# Access an object (e.g. a histogram) in the directory
+d.obj
+
+# ... or in a subdirectory
+d.subdir.obj
+
+# Wrong attribute: raises AttributeError
+d.wrongAttr
+\endcode
+
+Furthermore, TDirectory implements a `WriteObject` Python method which relies
```

Comment:
`WriteObject` is a no-op in `TDirectory` because `WriteObjectAny` is also a no-op in `TDirectory`, that's why I also mention `WriteObjectAny`. `TDirectoryFile` reimplements `WriteObjectAny`, so for `TDirectoryFile` and `TFile`, `WriteObject` actually does something.